httpsession/httpsession.go

```diff
@@ -0,0 +1,106 @@
+package httpsession
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/companieshouse/chs.go/log"
+	"github.com/companieshouse/go-session-handler/config"
+	session "github.com/companieshouse/go-session-handler/session"
+	"github.com/companieshouse/go-session-handler/state"
+	"github.com/justinas/alice"
+	redis "gopkg.in/redis.v5"
+)
+
+// Type for creating context keys
+type ContextKey string
+
+// Set the context key for the session
+var ContextKeySession = ContextKey("session")
+
+// Register will append an HTTP handler to an Alice chain, whereby the stored
+// session will be loaded and stored on the request context
+func Register(c alice.Chain) alice.Chain {
+	return c.Append(func(h http.Handler) http.Handler { return handler(h) })
+}
+
+// handler initialises a Store using config and cache structs, loads the
+// session, and stores it on the request context to access later
+func handler(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		// Init all config
+		cfg := config.Get()
+
+		redisOptions := &redis.Options{
+			Addr:     cfg.RedisServer,
+			DB:       cfg.RedisDB,
+			Password: cfg.RedisPassword,
```

Comment:
This is knowledge bleeding which cache we are using into the HTTP session code. Can you set up redis.Options struct inside of the cache please?